### PR TITLE
blog(add article): how to contribute to the otel demo

### PR DESCRIPTION
**Pull request by adnanrahic**
This PR adds an article that explains:

1. How to demystify OpenTelemetry by adding automatic and manual code
   instrumentation.
2. How to use trace-based testing in the OpenTelemetry Demo to maintain feature
   functionality and telemetry integrity.
3. How to contribute to the OpenTelemetry Demo safely while avoiding
   regressions.

### CODE REVIEW

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:290]: Can all of this be replaced with a reference to https://opentelemetry.io/docs/demo/ ? It duplicates a lot of words from there

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:301]: how is this fork different from upstream, when I follow that link I get `This branch is up to date with open-telemetry/opentelemetry-demo:main.`, also I would prefer you to use your personal github and not your org github for such a fork

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:358]: Can you make this (and other) code examples a little bit compacter? I think you can write more things in one line than you do it in the original code

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:440]: This belongs to the very top of the blog post, you should explain people what they are going to do. It's promised that they learn how to contribute, but the core of the blog post starts here.

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:446]: Let me be frank: to me it feels unnecessary to use Tracetest here. It totally makes sense on the tracetest blog, it also makes sense to talk about tracetest in the previous blog post https://opentelemetry.io/blog/2023/testing-otel-demo/, but this blog post is titled "How to contribute to the otel demo", so stick to that topic, aka use Jaeger for visualization

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:448]: Since you ask people above to take the code from a fork, this would be a place where you can shorten your blog post significantly. Make a version without the instrumentation and let people add it back in.

File: content/en/blog/2023/contributing-to-otel-demo/index.md
Comment (svrnm) [inline content/en/blog/2023/contributing-to-otel-demo/index.md:1021]: where did I learn that? I would have expected some words around raising a PR, etc

**Review by svrnm — CHANGES_REQUESTED** [review]
Thank you for suggesting this blog post, @adnanrahic. I appreciate the effort that you took to write it down, however there are some major concerns I have with the current form, I outlined them in my review comments already, but to summarize:

* The blog post suggest that it teaches you how to contribute to the otel demo, I don't see anything in the blog post around creating a PR, committing it, interacting with the community etc.
* The blog post makes unnecessary use of tracetest. Since tracetest is oss, there are cases where it can be used (like in https://opentelemetry.io/blog/2023/testing-otel-demo/), but it doesn't fit the title and content of the blog post. Please make use of Jaeger.
* The blog post is very long, so I made a lot of suggestions around reducing the size, mainly to skip a major part at the beginning explaining the demo, and that you start with a fork without the instrumentation and work from there.

**Comment by adnanrahic** [comment]
Thanks a lot for the very detailed review @svrnm !! 🙌 

You make great points. 🤝 

I've summarized a todo list to work on:

- [ ] Add a dedicated section about creating a PR, committing it, and interacting with the community.
- [ ] Add Jaeger to the use case for trace visualization instead of Tracetest. 
- [ ] Shorten the overall length of the article.
- [ ] Use the upstream repo (`open-telemetry/opentelemetry-demo:main`) as the base for explanations. _Note: I used the fork because some features were not yet merged into upstream at the time of writing. That can be changed now since they are merged.)_ 
- [ ] Use a fork in my personal GitHub account without instrumentation as the starting point of the tutorial and work from there.
- [ ] Move the `Demo X.` list of what the reader will learn to the top of the article.
- [ ] Make code blocks more compact and easy to read.
- [ ] Replace the `make start` section with a reference to `https://opentelemetry.io/docs/demo/` since there is a lot of content duplication.

**Comment by svrnm** [comment]
>     * [ ]  Add a dedicated section about creating a PR, committing it, and interacting with the community.

That or you rename your blog post to something more fitting 
 
>     * [ ]  Add Jaeger to the use case for trace visualization instead of Tracetest.

Exactly, since this is mainly about tracing, if you need something for metrics take prometheus

>     * [ ]  Shorten the overall length of the article.

Yes. And also try to keep the number of topics in the blog post "low". Right now there is a variety of things, that are only semi-related to the topic. Make cuts where possible. 

>     * [ ]  Use the upstream repo (`open-telemetry/opentelemetry-demo:main`) as the base for explanations. _Note: I used the fork because some features were not yet merged into upstream at the time of writing. That can be changed now since they are merged.)_

You can use your fork throughout the doc if it adds value, but don't send people to a fork that in sync with upstream.

>     * [ ]  Use a fork in my personal GitHub account without instrumentation as the starting point of the tutorial and work from there.

Exactly, ideally you put things in a special branch, since you might want to work with the demo in the future still.

>     * [ ]  Move the `Demo X.` list of what the reader will learn to the top of the article.

Don't see this as a purely mechanical advice: this is the *core* of what you are talking about in this blog post, think about how you can lead with that and what is not part of that can be removed.

>     * [ ]  Make code blocks more compact and easy to read.

Yes.

>     * [ ]  Replace the `make start` section with a reference to `https://opentelemetry.io/docs/demo/` since there is a lot of content duplication.

Keep the things that are necessary for the flow of the blog, but otherwise point to the docs, yes.

**Comment by svrnm** [comment]
@adnanrahic any updates?

**Comment by adnanrahic** [comment]
Hey @svrnm! 
Apologies for the delay on my end. I got accepted for a conference presentation and will get back to writing a new version of this once it is off my plate. 😬 
We are clear on the tasks, it's just up to me to get the free time. 🙌

**Review by chalin — CHANGES_REQUESTED** [review]
Drive-by comment that this post will need a copyedit pass or two once the major rework has landed. /cc @theletterf

**Comment by svrnm** [comment]
> Apologies for the delay on my end. I got accepted for a conference presentation and will get back to writing a new version of this once it is off my plate. 😬

Thanks, I'll close this PR for now and you can raise a new one whenever you are ready! This makes it easier for us maintainers to keep all the open PRs visible.

**Comment by adnanrahic** [comment]
Sounds good. Sorry for the delay. I should be able to pick it back up next week since my conf presentation is this weekend. 😄

**Pull request closed without merging**
